### PR TITLE
0.9.2

### PR DESCRIPTION
**Pull request by ericgaspar**
## Problem
- *Patch not working anymore

## Solution
- *And how you fix that*

## PR Status
- [ ] Code finished.
- [ ] Tested with Package_check.
- [ ] Fix or enhancement tested.
- [ ] Upgrade from last version tested.
- [ ] Can be reviewed and tested.

## Package_check results
---
* An automatic package_check will be launch at https://ci-apps-dev.yunohost.org/, when you add a specific comment to your Pull Request: "!testme", "!gogogadgetoci" or "By the power of systemd, I invoke The Great App CI to test this Pull Request!"*

### CODE REVIEW

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
I don't know how to test LDAP integration for Baïkal...

**Comment by ericgaspar** [comment]
!testme

**Comment by ericgaspar** [comment]
!testme

File: scripts/install [hunk 1]
```diff
@@ -77,87 +91,61 @@ ynh_app_setting_set --app=$app --key=final_path --value=$final_path
...
 
-#=================================================
-# SETUP SSOWAT
-#=================================================
-ynh_script_progression --message="Configuring permissions..." --weight=2
-
-# Allow public access on /
-ynh_permission_update --permission="main"  --add="visitors"
-# But restrain on /admin
-ynh_permission_create --permission="admin" --url="/admin" --allowed="all_users"
```
Comment (kay0u) [inline scripts/install:150]: Why did you remove the admin permission? (Also, you don't remove it in the update, so old baikal installations will have this permission while new ones won't)

Comment (ericgaspar) [reply scripts/install:150]: I was wondering why we need this permission...
I mean, isn't the `/admin` path  allowed to all users by default?

File: scripts/_common.sh [hunk 2]
```diff
@@ -5,7 +5,7 @@
 #=================================================
 
 # dependencies used by the app
-YNH_PHP_VERSION="7.3"
+YNH_PHP_VERSION="8.0"
 
 pkg_dependencies="php${YNH_PHP_VERSION}-xml php${YNH_PHP_VERSION}-mbstring php${YNH_PHP_VERSION}-mysql"
```
Comment (kay0u) [inline scripts/_common.sh:10]:
```suggestion
pkg_dependencies="php${YNH_PHP_VERSION}-xml php${YNH_PHP_VERSION}-mbstring php${YNH_PHP_VERSION}-mysql php${YNH_PHP_VERSION}-ldap"
```

File: conf/nginx.conf [hunk 3]
```diff
@@ -12,11 +12,6 @@ location __PATH__/ {
   # Path to source
```
Comment (kay0u) [inline conf/nginx.conf:12]: According to [this doc](https://sabre.io/baikal/upgrade/):

> If it was before something like:
> 
> http://example.org/baikal/cal.php
>
> Then it's now:
>
> http://example.org/baikal/html/dav.php
>
> So two things changed:
>
>    cal.php, card.php moved into the html/ directory.
>    We added dav.php. cal.php and card.php still exist in the html/ directory but will be removed in a future version. > dav.php is an endpoint for both caldav and carddav combined.

With our nginx conf we can still remove `html` from the url, but we must change
```nginx
location = /.well-known/carddav {
  return 301 https://$server_name__PATH__/card.php;
}
location = /.well-known/caldav {
  return 301 https://$server_name__PATH__/cal.php;
}
```

to:
```nginx
location = /.well-known/carddav {
  return 301 https://$server_name__PATH__/dav.php;
}
location = /.well-known/caldav {
  return 301 https://$server_name__PATH__/dav.php;
}
```

File: scripts/install (hunk 1 above)
Comment (kay0u) [reply scripts/install:150]: We need this permission because baikal must be public to visitors. And we want to protect `/admin`.

File: scripts/install [hunk 4]
```diff
@@ -77,87 +91,61 @@ ynh_app_setting_set --app=$app --key=final_path --value=$final_path
...
+chown $app: "$final_path/config/baikal.yaml"
 chmod 640 "$final_path/config/baikal.yaml"
 
-#=================================================
-# SETUP SSOWAT
-#=================================================
-ynh_script_progression --message="Configuring permissions..." --weight=2
-
-# Allow public access on /
-ynh_permission_update --permission="main"  --add="visitors"
```
Comment (kay0u) [inline scripts/install:148]: To be able to use baikal, please keep this line.


If you want to test baikal, you can install Thunderbird on the host of your ynh-dev, install baikal in your ynh-dev env.

In Thunderbird, click on `Menu -> New -> Calendar`, `Network`, then `your_username`, `Address: https://yourdomain-or-ip-of-your-ynhdev/baikal/`.
It will ask for the exception for the ssl certificate, you grant it, you put your user's password, and normally it works and you can add new tasks to your calendar

**Comment by ericgaspar** [comment]
!testme
